homeassistant/components/proxmoxve/config_flow.py

```diff
@@ -0,0 +1,188 @@
+            username = info.get(CONF_USERNAME, "")
+            password = info.get(CONF_PASSWORD, "")
+            realm = info.get(CONF_REALM, DEFAULT_REALM)
+            verify_ssl = info.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL)
+
+            if await self._async_endpoint_exists(f"{host}/{port}"):
+                _LOGGER.info(
+                    "Device %s already configured, you can remove it from configuration.yaml if you still have it",
+                    host,
+                )
+                return self.async_abort(reason="already_configured")
+
+            if port > 65535 or port <= 0:
+                errors[CONF_PORT] = "invalid_port"
+
+            if not errors:
+
+                try:
+                    self._proxmox_client = ProxmoxClient(
+                        host,
+                        port=port,
+                        user=username,
+                        realm=realm,
+                        password=password,
+                        verify_ssl=verify_ssl,
+                    )
+
+                    await self.hass.async_add_executor_job(
+                        self._proxmox_client.build_client
+                    )
+
+                except proxmoxer.backends.https.AuthenticationError:
+                    errors[CONF_USERNAME] = "auth_error"
+                except SSLError:
+                    errors[CONF_VERIFY_SSL] = "ssl_rejection"
+                except ConnectTimeout:
+                    errors[CONF_HOST] = "cant_connect"
+                except Exception:  # pylint: disable=broad-except
+                    errors["base"] = "general_error"
+
+                else:
+                    # get all vms and containers and add them
+
+                    if "nodes" not in self._config:
+                        self._config["nodes"] = []
+
+                    self._config[CONF_HOST] = host
+                    self._config[CONF_PORT] = port
+                    self._config[CONF_USERNAME] = username
+                    self._config[CONF_PASSWORD] = password
+                    self._config[CONF_REALM] = realm
+                    self._config[CONF_VERIFY_SSL] = verify_ssl
+
+                    await self.hass.async_add_executor_job(self._add_vms_to_config)
+
+                    return self.async_create_entry(title=host, data=self._config)
+
+            if errors and not is_import:
+                data_schema = vol.Schema(
+                    {
+                        vol.Required(CONF_HOST, default=info.get(CONF_HOST, "")): str,
+                        vol.Required(
+                            CONF_USERNAME, default=info.get(CONF_USERNAME, "")
+                        ): str,
+                        vol.Required(
+                            CONF_PASSWORD, default=info.get(CONF_PASSWORD, "")
+                        ): str,
+                        vol.Required(
+                            CONF_REALM, default=info.get(CONF_REALM, DEFAULT_REALM)
+                        ): str,
```

Comment:
the realm could be included as part of the username, so it is not strictly required as its own parameter

Reply:
Changed to optional.